src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/Diagnostics/QueueClientDiagnosticsTests.cs

```diff
@@ -6,558 +6,558 @@ namespace Microsoft.Azure.ServiceBus.UnitTests.Diagnostics
+                var eventQueue = this.CreateEventQueue();
 
-                this.listener.Enable((name, queue, arg) =>
-                    queueName == null || queue.ToString() != queueName);
-
-                await TestUtility.SendMessagesAsync(this.queueClient.InnerSender, 1);
-                this.queueClient.RegisterMessageHandler((msg, ct) =>
+                try
+                {
+                    using (var listener = this.CreateEventListener(queueName, eventQueue))
+                    using (var subscription = this.SubscribeToEvents(listener))
                     {
-                        processActivity = Activity.Current;
-                        processingDone.Set();
-                        return Task.CompletedTask;
-                    },
-                    exArgs => Task.CompletedTask);
-
-                processingDone.WaitOne(TimeSpan.FromSeconds(maxWaitSec));
-
-                Assert.True(this.events.IsEmpty);
-                Assert.Null(processActivity);
-            });
-        }
+                        listener.Enable((name, queue, arg) => !name.Contains("Receive") && !name.Contains("Exception"));
+                
+                        var parentActivity = new Activity("test").AddBaggage("k1", "v1").AddBaggage("k2", "v2");
+                
+                        parentActivity.Start();
+                        await TestUtility.SendSessionMessagesAsync(queueClient.InnerSender, 1, 1);
+                        parentActivity.Stop();
+                                
+                        var exceptionCalled = false;                
+                        var tcs = new TaskCompletionSource<Activity>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+                        queueClient.RegisterMessageHandler((msg, ct) =>
+                        {
+                            tcs.TrySetResult(Activity.Current);                    
+                            return Task.CompletedTask;
+                        },
+                        exArgs =>
+                        {
+                            // Do not set the completion source exception to avoid throwing 
+                            // when the task is awaited.  The sentinal variable is checked to detect
+                            // exception cases.
+                            exceptionCalled = true;                    
+                            return Task.CompletedTask;
+                        });
 
-        [Fact]
-        [LiveTest]
-        [DisplayTestMethodName]
-        async Task SendAndHandlerFireEvents()
-        {
-            await ServiceBusScope.UsingQueueAsync(partitioned: false, sessionEnabled: false, async queueName =>
-            {
-                this.queueClient = new QueueClient(TestUtility.NamespaceConnectionString, queueName, ReceiveMode.PeekLock);
+                        var processActivity = await tcs.Task.WithTimeout(TimeSpan.FromSeconds(MaxWaitSec));
```

Comment:
Having a `.WithDefaultTimeout()` overload might be useful, usually, people keep copying the same constant everywhere anyway.